### PR TITLE
Fix project plugin: only process comment if matches regex

### LINKED ISSUE #12117: prow/plugins/project: responds to every comment
**What happened**:
The `project` plugin responds to every comment, including:
- comments that don't match the regex
- comments from the bot itself

This created a comment loop.

**What you expected to happen**:
Plugin would not respond unless a comment from a real user matched the regex.

**How to reproduce it (as minimally and precisely as possible)**:
Enable plugin, comment on any issue.

**Please provide links to example occurrences, if any**:
https://github.com/kubernetes/test-infra/pull/12105

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by taragu**
Fixes #12117

/cc @stevekuznetsov 
/cc @cblecker

### CODE REVIEW

File: prow/plugins/project/project.go [hunk 1]
```diff
@@ -147,6 +147,9 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
 	if e.Action != github.GenericCommentActionCreated {
 		return nil
 	}
+	if !projectRegex.MatchString(e.Body) {
```
Comment (cblecker) [inline prow/plugins/project/project.go]: I'd suggest something like: https://github.com/kubernetes/test-infra/blob/c797b2df17692eff94d80f71b19e54cb20002838/prow/plugins/pony/pony.go#L127-L131

and then pass `mat` through into `parseCommand`. That way, you only really need to pass the entire body through a regex once.

Comment (taragu) [reply prow/plugins/project/project.go]: Thanks @cblecker ! Just made the update

**Comment by cblecker** [comment]
@taragu Can we also add a specific test case of a comment that doesn't match `/project` at all? Otherwise LGTM.

**Comment by taragu** [comment]
@cblecker sure! Just added the test case

**Comment by cblecker** [comment]
/lgtm
/approve
/hold

Will leave for @stevekuznetsov or @cjwagner to review as well.

**Comment by stevekuznetsov** [comment]
We also want to explicitly ignore comments from the robot itself, right?

**Comment by cblecker** [comment]
@stevekuznetsov It looks like only `sigmention` and `trigger` do that kind of validation, where as the rest of the plugins will act on any comment the bot makes. Is there *any* use case for the bot reacting to itself? Perhaps that validation would better live upstream of the plugins themselves.

**Comment by cblecker** [comment]
@taragu Yeah, let's go ahead and add one more safeguard against the bot reacting to itself. We're still not sure if this should be an across the board thing, but we want to tread a little lighter.

https://github.com/kubernetes/test-infra/blob/86a523febb3c6af836859dda5624dcbea7204f7a/prow/plugins/sigmention/sigmention.go#L79-L85

File: prow/plugins/project/project.go [hunk 2]
```diff
@@ -147,10 +141,21 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
 	if e.Action != github.GenericCommentActionCreated {
```
Comment (cblecker) [inline prow/plugins/project/project.go]: // Only handle new comments

File: prow/plugins/project/project.go [hunk 3]
```diff
@@ -147,10 +141,21 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
...
 		return nil
 	}
+	botName, err := gc.BotName()
```
Comment (cblecker) [inline prow/plugins/project/project.go]: // Only handle comments that don't come from the bot

File: prow/plugins/project/project.go [hunk 4]
```diff
@@ -147,10 +141,21 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
...
+	if err != nil {
+		return err
+	}
+	if e.User.Login == botName {
+		return nil
+	}
+	matches := projectRegex.FindStringSubmatch(e.Body)
```
Comment (cblecker) [inline prow/plugins/project/project.go]: // Only handle comments that match the regex

**Comment by taragu** [comment]
@cblecker sounds good. Just make the update

**Comment by cblecker** [comment]
/lgtm
/approve

/assign @stevekuznetsov 
for approval

**Pull request merged**